### ISSUE TITLE
Feature Request: Order elements to Group Paths

### ISSUE THREAD

**Issue by mr-CKR** [svg/svgo#941]
The elements could be reorderd to save path elements

**Sample SVG**
```
<svg>
    <path d="M0,0 h25 v25 h-25 z"/>
    <use href="#a"/>
    <path d="M100,100 h25 v25 h-25 z"/>
<svg/>
```

**Output**
```
<svg>
    <use href="#a"/>
    <path d="M0,0 h25 v25 h-25 z M100,100 h25 v25 h-25 z"/>
<svg/>
```

**Comment by GreLI** [comment]
Yes, it's possible and the idea is quite trivial. However, figures may overlap and you must be sure that that is not the case. And that is where an implementation become not trivial, because 1) there is no fast algorithm for detecting intersection that I'm aware of, 2) you must consider strokes, transformations and all that stuff which complicates things a lot. Please, let me know if you aware of solutions for these problems.

**Comment by mr-CKR** [comment]
1) I'm not aware of any solution for this, but I might just dive into it to see to what extent it is possible.

2) Don't these problems already exist with current path merger operations? And a possible workaround would be to only apply this when no styling exists at all.
